### PR TITLE
Cut v2.12.0-rc.0

### PR DESCRIPTION
**Pull request by juliusv**
Signed-off-by: Julius Volz <julius.volz@gmail.com>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,27 @@
+## 2.12.0-rc.0
+
+* [FEATURE] Add PromQL query logging. #5794
+* [FEATURE] Enable and provide binaries for `mips64` / `mips64le` architectures. #5792
+* [ENHANCEMENT] Mark deleted rule's series as stale on next evaluation. #5759
+* [ENHANCEMENT] Check for duplicate label names in remote read. #5829
```
Comment (juliusv) [inline CHANGELOG.md]: This could also be seen as either BUGFIX or CHANGE, not sure?

Comment (brian-brazil) [reply CHANGELOG.md]: Bugfix

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,27 @@
+## 2.12.0-rc.0
+
+* [FEATURE] Add PromQL query logging. #5794
```
Comment (brian-brazil) [inline CHANGELOG.md]: Active queries. This is showing you what was running when a prometheus crashed.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,27 @@
...
+* [FEATURE] Enable and provide binaries for `mips64` / `mips64le` architectures. #5792
+* [ENHANCEMENT] Mark deleted rule's series as stale on next evaluation. #5759
```
Comment (brian-brazil) [inline CHANGELOG.md]: Bugfix?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,27 @@
...
+* [ENHANCEMENT] Check for duplicate label names in remote read. #5829
+* [ENHANCEMENT] Add more details about retention to storage docs. #5842
```
Comment (brian-brazil) [inline CHANGELOG.md]: Docs don't get mentioned, they're not part of the binary

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,27 @@
...
+* [ENHANCEMENT] Improve responsiveness of targets web UI and API endpoint. #5740
+* [ENHANCEMENT] Improve remote write desired shards calculation. #5763
+* [ENHANCEMENT] Flush TSDB pages more precisely. tsdb#660
+* [ENHANCEMENT] Add `prometheus_tsdb_retention_limit_bytes` metric. tsdb#667
+* [ENHANCEMENT] Add logging during TSDB WAL replay on startup. tsdb#662
+* [ENHANCEMENT] Improve TSDB memory usage. tsdb#653, tsdb#643, tsdb#654, tsdb#642, tsdb#627
+* [BUGFIX] Fix JavaScript error when showing warning about out-of-sync server time. #5833
+* [BUGFIX] Fix example command lines for starting `remote_write_adapter` in docs. #5830
```
Comment (brian-brazil) [inline CHANGELOG.md]: Ditto

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,27 @@
...
+* [BUGFIX] Correct function name in remote write queue log message. #5852
```
Comment (brian-brazil) [inline CHANGELOG.md]: This is too small to mention I'd say, plus a few of the others. It doesn't affect the server's semantics.

**Comment by codesome** [comment]
I guess this should go in after https://github.com/prometheus/prometheus/pull/5892 v2.11.2 is cut and merged back into master?

**Comment by juliusv** [comment]
@brian-brazil Addressed comments and deleted a bunch of items.

**Comment by juliusv** [comment]
@codesome yes, will rebase ontop of that once that is merged

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,21 @@
...
+* [FEATURE] Track currently active PromQL queries in a log file. #5794
+* [FEATURE] Enable and provide binaries for `mips64` / `mips64le` architectures. #5792
+* [ENHANCEMENT] Improve responsiveness of targets web UI and API endpoint. #5740
+* [ENHANCEMENT] Improve remote write desired shards calculation. #5763
+* [ENHANCEMENT] Flush TSDB pages more precisely. tsdb#660
+* [ENHANCEMENT] Add `prometheus_tsdb_retention_limit_bytes` metric. tsdb#667
+* [ENHANCEMENT] Add logging during TSDB WAL replay on startup. tsdb#662
+* [ENHANCEMENT] Improve TSDB memory usage. tsdb#653, tsdb#643, tsdb#654, tsdb#642, tsdb#627
+* [BUGFIX] Check for duplicate label names in remote read. #5829
+* [BUGFIX] Mark deleted rule's series as stale on next evaluation. #5759
```
Comment (brian-brazil) [inline CHANGELOG.md]: rules'

Comment (juliusv) [reply CHANGELOG.md]: Haha I was debating whether I should fix / change that :P Done.

**Pull request merged**